scenario.py

```diff
@@ -211,6 +227,10 @@ def __init__(self, params, experiment_path, scenario_id=1, n_repeat=1, is_dry_ru
             self.epoch_count = 3
             self.minibatch_count = 2
 
+        if self.proportion_dataset < 1:
+            self.shorten_dataset_proportion()
```

Comment:
Just to be sure: have you checked if the other methods based on the dataset nb of samples (some asserts to verify if mini-batches will be possible etc.) are triggered after that one and based on the updated nb of samples after the truncation?

Reply:
Currently there is a conflict with:
https://github.com/SubstraFoundation/distributed-learning-contributivity/blob/f3c79f1fc5064ec9fd4638fbd01ed345f68e3df3/scenario.py#L74-L78

Reply:
I have solved that